dev-support/ranger-docker/docker-compose.ranger.yml

```diff
@@ -8,7 +8,7 @@ services:
         - RANGER_VERSION=${RANGER_VERSION}
         - RANGER_DB_TYPE=${RANGER_DB_TYPE}
         - RANGER_ADMIN_JAVA_VERSION=${RANGER_ADMIN_JAVA_VERSION}
-    image: ranger:latest
+    image: ranger:2.4.0
```

Comment:
does this mean we are using the official 2.4.0 docker image in our docker containers?  or are we using the image we are building?  if it is the one we are building, i think we should give it a different name, something like 2.4.0-with-hmsa.

Reply:
No, this isn't the official image. Check the description here https://hub.docker.com/r/apache/ranger

To get the official image from dockerhub, we need to replace `ranger:2.4.0` with `apache/ranger:2.4.0`.

Just replacing `latest` with `2.4.0` doesn't do what we expect it to. I think that we are running ranger from sources. Let me explain and correct me if I'm wrong. This is related to the issue that we had with @mladjan-gadzic when starting ranger in docker for the first time with just the ranger and postgres compose files.

This `docker-compose.ranger.yml` file is using `Dockerfile.ranger`.

In the `Dockerfile.ranger`, on top there is this line

```
FROM ranger-base:latest AS ranger
```
https://github.com/apache/ranger/blob/master/dev-support/ranger-docker/Dockerfile.ranger#L18

So just `ranger` and not `apache/ranger` points to `ranger-base`.

`ranger-base` has to be built and cached locally like so

```
> docker-compose -f docker-compose.ranger-base.yml build --no-cache
```

`ranger-base` downloads dependencies and sets environment variables. We are building `ranger-base` so that we can prepare the env and execute `ranger-build` which depends on it.

`ranger-build` copies and sets up all the necessary files under the containers.

```
> docker-compose -f docker-compose.ranger-base.yml -f docker-compose.ranger-build.yml down --remove-orphans

> docker-compose -f docker-compose.ranger-base.yml -f docker-compose.ranger-build.yml up
```

Under `dev-support/ranger-docker/dist` there are all the package tarballs that were generated from the build. `ranger-build` sets them up under the containers. Then the tarball files are used for running ranger. That's why I think that we are running ranger from sources.

Our workaround has been to run `./ranger_in_docker up` once because it takes care of `ranger-base` and `ranger-build`.